application/plugins/external_script/external_script.php

```diff
@@ -85,68 +63,75 @@ function external_script_install()
 
 function external_script($sms)
 {
-	$config = external_script_initialize();
+	$scripts = external_script_initialize();
 	$phone = $sms->SenderNumber;
 	$content = $sms->TextDecoded;
 	$id = $sms->ID;
 	$time = $sms->ReceivingDateTime;
-	$shell_path = $config['ext_script_path'];
-			
-	// Load all rules	
-	foreach($config['ext_script'] as $rule)
+	$match = NULL; //The result of a preg_match capture
+
+	// Load all rules
+	foreach($scripts as $rule)
 	{
-		$script_name = $rule['name'];
+		$intepreter_path = $rule['intepreter_path'];
+		$script_path = $rule['script_path'];
 		$value=$parameter="";
-		
+
 		// evaluate rule key
 		switch($rule['key'])
 		{
 			case 'sender':
 				$value = $phone;
 			break;
-			
+
 			case 'content':
 				$value = $content;
 			break;
 		}
-		
+
 		// evaluate rule type
 		switch($rule['type'])
 		{
-			case 'match':
-				$is_valid = is_match($rule['value'], $value);
+			case 'equal':
```

Comment:
IMO we should also keep `match` as to not break anyone's current setup.

Reply:
I renamed it to be more in line with what it really does.
Do you think someone may use that function from somewhere else that in this file ?
I'm just asking, because since I moved the config of the scripts into "config" subdir, people using that plugin will have to do some slight rework anyway.

But if you prefer I rename it back (and it will have a name close to the new "preg_match" function).

Reply:
No, I meant we should also add `case 'match':` as equal to `case 'equal':`  so as to be backward compatible. This would give folks time to migrate and not break their current setup.

Reply:
changed